src/ComputeSharp.D2D1.SourceGenerators/Diagnostics/Analyzers/InvalidD2DInputArgumentAnalyzer.cs

```diff
@@ -0,0 +1,166 @@
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using ComputeSharp.D2D1.Interop;
+using ComputeSharp.D2D1.Intrinsics;
+using ComputeSharp.SourceGeneration.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using static ComputeSharp.SourceGeneration.Diagnostics.DiagnosticDescriptors;
+
+namespace ComputeSharp.D2D1.SourceGenerators;
+
+/// <summary>
+/// A diagnostic analyzer that generates an error whenever [D2DEffectId] is used with an invalid value.
```

Comment:
`generates an error whenever [D2DEffectId]...`

copypasta? Should be `generates an error whenever D2D.GetInput() ...`, right?

Reply:
Yup ahahah